test/performance/odp_sched_pktio.c

```diff
@@ -238,6 +238,26 @@ static int worker_thread_direct(void *arg)
 			test_global->worker_stat[worker_id].rx_pkt += num_pkt;
 	}
 
+	/*
+	 * Free prefetched packets before exiting worker thread as
+	 * such packets can block sending of later packets in other threads.
+	 */
+	odp_schedule_pause();
+	while (1) {
+		odp_event_t  ev;
+
+		ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
+		if (ev == ODP_EVENT_INVALID)
+			break;
+		odp_event_free(ev);
+	}
+
+	/* Flush all writes */
+	odp_mb_full();
+
+	/* Restore scheduling state */
+	odp_schedule_resume();
```

Comment:
So after this odp_schedule_resume() call there can again be local prefetched events and we are back to square one? I do not see how this patch fixes something, other than by chance through altering timing.

Maybe there should be thread barrier (like in l2fwd) to make sure everybody stops sending packets before anybody resumes scheduling again for the final flush (if it is needed anymore then). Or something like that.

Reply:
Removed odp_schedule_resume() in V2. Let me know if it makes sense given that main thread is cleaning up all the other events in scheduler so I believe odp_l2fwd like per thread cleanup is not needed.

Reply:
After the removal of the schedule_resume() this looks ok to me, apart from the minor unrelated comments I made.